src/runtime_src/hip/api/hip_module.cpp

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023-2024 Advanced Micro Device, Inc. All rights reserved.
+
+#include "core/common/error.h"
+
+#include "hip/config.h"
+#include "hip/hip_runtime_api.h"
+
+#include "hip/core/module.h"
+
+namespace xrt::core::hip {
+static void
+hipModuleLaunchKernel(hipFunction_t f, uint32_t gridDimX, uint32_t gridDimY,
```

Comment:
I am curious about which device will be targeted. FPGA with HLS or OpenCL?

Reply:
it can be used for different kind of devices - alveo, ryzen ai etc